presto-tests/src/main/java/com/facebook/presto/tests/DistributedQueryRunner.java

```diff
@@ -269,4 +273,14 @@ public final void close()
             throw Throwables.propagate(e);
         }
     }
+
+    private void cancelAllQueries()
+    {
+        QueryManager queryManager = coordinator.getQueryManager();
+        for (QueryInfo queryInfo : queryManager.getAllQueryInfo()) {
+            if (!queryInfo.getState().isDone()) {
+                queryManager.cancelQuery(queryInfo.getQueryId());
+            }
+        }
```

Comment:
Since cancels are async, I would add a loop that spins until they're all in a "done" state (or until a second or two have passed)